libsolidity/ast/Types.cpp

```diff
@@ -1413,14 +1413,6 @@ bool ContractType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 	return false;
 }
 
-bool ContractType::isExplicitlyConvertibleTo(Type const& _convertTo) const
-{
-	return
-		isImplicitlyConvertibleTo(_convertTo) ||
-		_convertTo.category() == Category::Integer ||
```

Comment:
Maybe we should add a test for explicit contract-to-integer conversions? Do we want to disable this here? Explicit integer-to-contract conversions are still allowed in this PR...

Reply:
Erm, this is weird. I'll have to re-check the PR.

The following should be possible:

 - implicit conversions between derived and base contract type, no other implicit conversions from or to contracts.
 - explicit conversions between contracts and addresses (in both directions).

Reply:
:smile: